packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -298,6 +300,14 @@ export async function processHTTPRequest<TContext>(
         parseGraphQLRequest(httpRequest.request, requestParams),
       );
 
+      if (options.allowBatchedHttpRequests === false) {
```

Comment:
I'm going to move this up a few lines — no use parsing the request if we're going to immediately throw.